AIPrompt docs (#1883)
telerik/blazor-docs: components/aiprompt/views/prompt.md

---
title: Prompt View

> page_title: AIPrompt Prompt View

Reviewer: ```suggestion
page_title: AIPrompt - Prompt View
```

description: Explore the AIPrompt Prompt View.
slug: aiprompt-views-prompt
tags: telerik,blazor,aiprompt,prompt
published: True
position: 10
---

# AIPrompt Prompt View

> The Prompt View features the prompt input, where the user can type in their query. It also contains a button, which when pressed, triggers a response request.

Reviewer 2: ```suggestion
The Prompt View features the prompt input, where users can type their query. It also contains a button to trigger a response request.
```

Additionally, the Prompt View can contain prompt suggestions related to the prompt itself through the `PromptSuggestions` parameter. The user can select any of the available suggestions, which in turn will populate the prompt input with the selected suggestion. This interaction will not trigger a response request right away - the user can modify the suggestion first.

> >caption Use the `PromptSuggestions` to display a set of predefined prompts or hints.

Reviewer: ```suggestion
>caption Using `PromptSuggestions` to display a set of predefined prompts or hints
```

````CSHTML
<TelerikAIPrompt OnPromptRequest="@HandlePromptRequest" PromptSuggestions="PromptSuggestions"></TelerikAIPrompt>

@code {
    private List<string> PromptSuggestions { get; set; } = new List<string>()
    {
        "Act as a marketing specialist and content writer and write a compelling [type of text] that speaks directly to my [ideal customer persona] and encourages them to take [desired action] on my [website/product].",
        "I'm looking for a [type of text] that will convince [ideal customer persona] to sign up for my [program/subscription] by explaining the value it brings and the benefits they'll receive.",
        "Write a Twitter thread idea that will both go viral and attract high-quality leads for my [product/service] with a strong call-to-action and compelling visuals."
    };

    private void HandlePromptRequest(AIPromptPromptRequestEventArgs args)
    {

>         // dummy data intentionally used. Replace the hard-coded string with a call to your AI API.

Reviewer: ```suggestion
        // This example uses dummy data intentionally. Replace the hard-coded string with a call to your AI API.
```
